agent/connect/uri_signing.go

```diff
@@ -64,6 +62,6 @@ func (id SpiffeIDSigning) CanSign(cu CertURI) bool {
 // break all certificate validation. That does mean that DNS prefix might not
 // match the identity URIs and so the trust domain might not actually resolve
 // which we would like but don't actually need.
-func SpiffeIDSigningForCluster(config *structs.CAConfiguration) *SpiffeIDSigning {
-	return &SpiffeIDSigning{ClusterID: config.ClusterID, Domain: "consul"}
+func SpiffeIDSigningForCluster(clusterID string) *SpiffeIDSigning {
```

Comment:
Generally I do prefer accepting a struct instead of a string arg, but in this case I think the string arg is better because:
1. this `structs.CAConfiguration` is part of a public interface, so its use here is being overloaded
2. the `ClusterID` is not actually part of the CAConfiguration. It's not settable by the user, and will likely be moved off of that struct.

If we ever need more args to `SpiffeIDSigningForCluster` we should create a separate struct for it.